app/services/order_details/param_updater.rb

```diff
@@ -70,6 +70,7 @@ def update_attributes(params)
         change_order_status(order_status_id, @options[:cancel_fee]) || raise(ActiveRecord::Rollback)
       else
         @order_detail.save_as_user(@editing_user) || raise(ActiveRecord::Rollback)
+        LogEvent.log(@order_detail, :price_change, @editing_user) unless @order_detail.price_changed_by_user == nil
```

Comment:
Will we get duplicate logs if: 
* User 1 makes a price change. This should log as expected.
* User 2 makes a non-price change

Will this second action also end up with a log because the price changed by user is user 1?

Maybe we could use something in [ActiveRecord::Dirty](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-saved_change_to_attribute-3F) like `saved_change_to_price_change_reason?`

Or maybe even simpler: move the log back into `assign_price_changed_by_user`, but move the call to that method _inside_ the transaction block. That way if the save on order detail fails, it raises the rollback and the log event creation should get rolled back as well.